### PR TITLE
fix: enable shader variants prewarm

### PR DESCRIPTION
**Pull request by pravusjif**
* Created shader variants collection files for all shaders (not used as it takes forever to compile when the explorer starts)
* Created shader variants collection with a few common selected shaders and variants, added a prewarm call for that when the application starts
* Disabled 'optimize mesh data' in player settings as it's been reported that it causes problems when switching materials and shaders in runtime, besides making builds take longer when compiling shader variant files.

If the shaders prewarm time is unacceptable we'll have to implemenet a new event from kernel telling the renderer when the loading will begin (or the play button was clicked) and move the shaders warmup to that point

### CODE REVIEW

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
Testing the build I noticed that some materials in my avatar are being drawed as blank:

image

**Review by sandrade-dcl — APPROVED** [review]

**Comment by sandrade-dcl** [comment]
> Testing the build I noticed that some materials in my avatar are being drawed as blank:
> 
> image

Solved!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs [hunk 1]
```diff
@@ -45,6 +45,9 @@ void Awake()
             Debug.unityLogger.logEnabled = false;
 #endif
 
+            // Warmup some shader variants
+            Resources.Load<ShaderVariantCollection>("ShaderVariantCollections/shaderVariants-selected").WarmUp();
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs]: If this call is blocking, this means the kernel messages can start pouring in before the rest of the code in this `Awake()` call, leading to issues. We should move this call at the very least to the end of this method.

**Review by BrianAmadori — CHANGES_REQUESTED** [review]
The compiling delay is quite noticeable. In my case, the freeze could be observed *after* the rendering was activated.

Options over the top of my head:
- Try to narrow the variants in the shadervariants file even further
- Try to ensure the prewarm happens before the rendering is activated, or block the rendering ack until the prewarm is finished
- Try to bundle the variants inside some asset bundle file and see if it takes less time.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs [hunk 2]
```diff
@@ -222,99 +225,99 @@ public bool ProcessMessage(MessagingBus.QueuedSceneMessage_Scene msgObject, out
...
-                    }
+                            break;
+                        }
                     case MessagingTypes.ENTITY_REPARENT:
-                    {
-                        if (msgObject.payload is Protocol.SetEntityParent payload)
-                            scene.SetEntityParent(payload.entityId, payload.parentId);
+                        {
+                            if (msgObject.payload is Protocol.SetEntityParent payload)
+                                scene.SetEntityParent(payload.entityId, payload.parentId);
```
Comment (AjimenezDCL) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs:199]: Any reason for this extra indentation?

**Review by AjimenezDCL — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs (hunk 2 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs:199]: the auto formatter file we have in the repo should be responsible for that, probably we have 2 different ones, one for VSCode and another for Rider? or where do you guys store your formatting rules?

Comment (AjimenezDCL) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs:199]: We dont have a common one, we should though. @BrianAmadori

**Comment by pravusjif** [comment]
@BrianAmadori I moved the warmup next to the pools warmup in `SceneController.Start()` and now I don't notice the stall at all, check it out

**Review by BrianAmadori — APPROVED** [review]
Tested it in incognito + cache disabled and the stall is no more. Marking as approved!

**Pull request merged**